File: 16_bit_manipulation/04_missing_number/missing_number.py
from typing import List


class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        exists: List[bool] = [False] * (len(nums) + 1)
        for num in nums:
            exists[num] = True
        for num in range(len(exists)):
            if exists[num] is False:
                return num
        return -1


if __name__ == "__main__":
    solution = Solution()
    print(solution.missingNumber([3, 0, 1]))

File: NeetCode/16_bit_manipulation/04_missing_number/missing_number.cpp
#include <vector>

using namespace std;

class Solution
{
  public:
    int missingNumber(vector<int>& nums)
    {
        vector<bool> exists(nums.size() + 1, false);
        for (int num : nums)
            exists[num] = true;
        for (int num = 0; num < exists.size(); ++num)
        {
            if (!exists[num])
                return num;
        }
        return -1;
    }
};